smack-core/src/main/java/org/jivesoftware/smack/proxy/ProxyInfo.java

```diff
@@ -74,6 +70,18 @@ public static ProxyInfo forSocks5Proxy(String pHost, int pPort, String pUser,
         return new ProxyInfo(ProxyType.SOCKS5, pHost, pPort, pUser, pPass);
     }
 
+    public Proxy.Type getJavaProxyType() {
+        switch (proxyType) {
+            case HTTP:
+                return Proxy.Type.HTTP;
+            case SOCKS4:
+            case SOCKS5:
+                return Proxy.Type.SOCKS;
+            default:
+                throw new AssertionError("Unreachable case.");
```

Comment:
Exception messages should always strive to include sensible context, e.g. in this case `"Unkown proxy type: "  + proxyType`. But since you cover all cases of theProxyInfo.ProxyType enum, do you need a default case at all?